File: backend/src/EnterpriseAuth.Api/Core/Domain/Entities/SalesInvoiceCustomer.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EnterpriseAuth.Api.Core.Domain.Entities
{
    [Table("SalesInvoiceCustomers")]
    public class SalesInvoiceCustomer
    {
        [Key]
        [MaxLength(100)]
        [Column("Code")]
        public string Code { get; set; } = string.Empty;

        [Required]
        [MaxLength(255)]
        [Column("Name")]
        public string Name { get; set; } = string.Empty;

        [MaxLength(50)]
        [Column("PaymentTerm")]
        public string? PaymentTerm { get; set; }

        [Column("CreditLimit", TypeName = "decimal(18,2)")]
        public decimal? CreditLimit { get; set; }

        [Column("StatusFlag")]
        public int StatusFlag { get; set; }

        [Column("IsProcessed")]
        public bool IsProcessed { get; set; }

        [Column("CreatedAt")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("UpdatedAt")]
        public DateTime? UpdatedAt { get; set; }
    }
}

File: frontend/lib/features/logistics/domain/entities/sales_invoice_customer.dart
class SalesInvoiceCustomer {
  final String code;
  final String name;
  final String? paymentTerm;
  final double? creditLimit;
  final int? statusFlag;
  final int isSynced;

  SalesInvoiceCustomer({
    required this.code,
    required this.name,
    this.paymentTerm,
    this.creditLimit,
    this.statusFlag,
    this.isSynced = 1,
  });

  factory SalesInvoiceCustomer.fromJson(Map<String, dynamic> json) {
    return SalesInvoiceCustomer(
      code: json['code'] ?? '',
      name: json['name'] ?? '',
      paymentTerm: json['paymentTerm'],
      creditLimit: json['creditLimit'] != null ? (json['creditLimit'] as num).toDouble() : null,
      statusFlag: json['statusFlag'] != null ? int.tryParse(json['statusFlag'].toString()) : null,
      isSynced: json['isSynced'] ?? 1,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'code': code,
      'name': name,
      'paymentTerm': paymentTerm,
      'creditLimit': creditLimit,
      'statusFlag': statusFlag,
      'isSynced': isSynced,
    };
  }
}
